src/gps.cpp

```diff
@@ -0,0 +1,91 @@
+#include "gps.hpp"
+
+#include <Arduino.h>
+
+namespace obc {
+
+constexpr auto gps_baud_rate = 9600L;
+constexpr auto gps_delay = 1000L;
+void init(Gps& gps)
+{
+    gps.begin(gps_baud_rate);
+    gps.sendCommand(PMTK_SET_NMEA_OUTPUT_RMCGGA);
+    gps.sendCommand(PMTK_SET_NMEA_UPDATE_1HZ);  // 1 Hz update rate
+    gps.sendCommand(PGCMD_ANTENNA);
+    delay(gps_delay);
+    Serial3.println(PMTK_Q_RELEASE);
+}
+GpsMeasurments measure_gps(Gps& gps)
+{
+    char c = gps.read();
+    if (GPSECHO)
+        if (gps.read() != 0) Serial.print(c);
+    if (gps.newNMEAreceived()) {
+        if (!gps.parse(gps.lastNMEA())) return {{}, {}, {}, Error::Busy};
+    }
+    return {
+        {gps.hour, gps.minute, gps.seconds, gps.milliseconds},
+        {gps.year, gps.month, gps.day},
+        {gps.fix,
+         gps.fixquality,
+         gps.longitudeDegrees,
+         gps.lon,
+         gps.latitudeDegrees,
+         gps.lat,
+         gps.altitude,
+         gps.speed,
+         gps.satellites},
+        Error::Ok};
+}
+void print(const GpsTime& gps_time)
+{
+    Serial.print("\nTime: ");
+    if (gps_time.hour < format_helper[0]) { Serial.print('0'); }
```

Comment:
ten format_helper jest strasznie niejasny. Powinieneś ponazywać każdą ze stałych osobno, tak żeby było od razu wiadomo o co chodzi.

alternatywnie, mozna zrobić coś takiego
```c++
template <typename T> // requires std::unsigned_integral<T>
bool has_tens_digit(T n) { return n > 9; }

template <typename T> // requires std::unsigned_integral<T>
bool has_hundreds_digit(T n) { return n > 99; }

//-- snip! --
if (!has_tens_digit(n)) { print('0'); }
//-- snip! --
```
zwróc na te komentarze z ```requires```. One są ważne bo dokumentują wymagania, które dany T musi spełniać zeby funkcja działała poprawnie. unsigned_integral to jest _koncept_.